api/v1beta1/azuremachinetemplate_webhook.go

```diff
@@ -92,5 +104,9 @@ func (r *AzureMachineTemplate) ValidateDelete() error {
 
 // Default implements webhookutil.defaulter so a webhook will be registered for the type.
 func (r *AzureMachineTemplate) Default() {
-	r.Spec.Template.Spec.SetDefaults()
```

Comment:
Rather than changing this here, shouldn't we update the `func (s *AzureMachineSpec) SetDefaults()` method directly in `api/v1beta1/azuremachine_default.go`? That `SetDefaults()` method is also invoked in `api/v1beta1/azuremachine_webhook.go` (in the `func (m *AzureMachine) Default()` method)

Reply:
I think it makes sense to default the `RoleAssignmentName` to a random UUID in the `AzureMachine` object, that's fine because all VMs would have a different one. It only doesn't make sense at the `AzureMachineTemplate` level.

Reply:
Thanks for the explanation, makes sense